pkg/publisher/publisher.go

```diff
@@ -207,6 +198,24 @@ func (p *cloudWatchPublisher) getCloudWatchMetricNamespace() *string {
 	return aws.String(cloudwatchMetricNamespace)
 }
 
+func getClusterId(ec2Client *ec2wrapper.EC2Wrapper) string {
+	clusterID, err := ec2Client.GetClusterTag("eks:cluster-name")
+	if err != nil || clusterID == "" {
+		glog.Errorf("Failed to obtain eks:cluster-name, fetching CLUSTER_ID.  %v", err)
+		clusterID, err := ec2Client.GetClusterTag("CLUSTER_ID")
+		if err != nil || clusterID == "" {
+			glog.Errorf("Failed to obtain CLUSTER_ID, fetching Name.  %v", err)
+			clusterID, err = ec2Client.GetClusterTag("Name")
+			if err != nil || clusterID == "" {
+				glog.Errorf("Failed to determine cluster-id, defaulting to 'k8s-cluster'.  %v", err)
+				clusterID = "k8s-cluster"
+			}
+		}
+	}
+	glog.Info("Using cluster ID ", clusterID)
+	return clusterID
+}
```

Comment:
Hi @groodt thanks very much for the contribution! May I suggest reworking the above to be a little cleaner? You can use a const to list the tag names to try, in order, and then loop through the tag names like so:

```go
const (
        defaultClusterID = "k8s-cluster"
        clusterIDTags = []string{
                "eks:cluster-name",
                "CLUSTER_ID",
                "Name",
        }
)
...
func getClusterID(ec2Client *ec2wrapper.EC2Wrapper) string {
        var clusterID string
        var err error
        for _, tag := range clusterIDTags {
	        clusterID, err = ec2Client.GetClusterTag(tag)
	        if err != nil && clusterID != "" {
                        break
		}
	}
        if clusterID == "" {
                clusterID = defaultClusterID
        }
	glog.Info("Using cluster ID ", clusterID)
	return clusterID
}
```

Reply:
Thanks for the suggestion. That is cleaner. I'll make the changes ASAP.

Reply:
I've made the change. Slices can't be `const`, so I've inlined the variables into the function. Open to alternatives. Just let me know what would be the preference.

Reply:
Yes, forgot about that. You could have either just replaced `const` with `var`, but this is perfectly fine as well :)